scipy/sparse/linalg/tests/test_pydata_sparse.py

```diff
@@ -0,0 +1,226 @@
+import pytest
+
+import numpy as np
+import scipy.sparse as sp
+import scipy.sparse.linalg as splin
+
+from numpy.testing import assert_allclose
+
+try:
+    import sparse
+except ImportError:
+    sparse = None
+
+pytestmark = pytest.mark.skipif(sparse is None,
+                                reason="pydata/sparse not installed")
+
+
+def test_isolve_gmres():
+    # Several of the iterative solvers use the same
+    # isolve.utils.make_system wrapper code, so test just one of them.
+    np.random.seed(1234)
+
+    A = sparse.COO(np.random.rand(5, 5))
+    b = np.random.rand(5)
+
+    x, info = splin.gmres(A, b, atol=1e-15)
+    assert info == 0
+    assert isinstance(x, np.ndarray)
+    assert_allclose(A @ x, b)
+
+
+def test_lsmr():
+    np.random.seed(1234)
+
+    A_dense = np.random.rand(5, 5)
+    A = sparse.COO(A_dense)
+    b = np.random.rand(5)
+
+    res0 = splin.lsmr(A_dense, b)
+    res = splin.lsmr(A, b)
+    assert_allclose(res[0], res0[0])
+
+
+def test_lsqr():
+    np.random.seed(1234)
+
+    A_dense = np.random.rand(5, 5)
+    A = sparse.COO(A_dense)
+    b = np.random.rand(5)
+
+    res0 = splin.lsqr(A_dense, b)
+    res = splin.lsqr(A, b)
+    assert_allclose(res[0], res0[0])
+
+
+def test_eigs():
+    np.random.seed(1234)
+
+    A_dense = np.random.rand(10, 10)
+    A_dense = A_dense @ A_dense.T
+    A = sparse.COO(A_dense)
+
+    M_dense = np.random.rand(10, 10)
+    M_dense = M_dense @ M_dense.T
+    M = sparse.COO(M_dense)
+
+    v0 = np.random.rand(10)
+
+    w_dense, v_dense = splin.eigs(A_dense, k=3, v0=v0)
+    w, v = splin.eigs(A, k=3, v0=v0)
+
+    assert_allclose(w, w_dense)
+    assert_allclose(v, v_dense)
+
+    w_dense, v_dense = splin.eigs(A_dense, M=M_dense, k=3, v0=v0)
+    w, v = splin.eigs(A, M=M_dense, k=3, v0=v0)
+
+    assert_allclose(w, w_dense)
+    assert_allclose(v, v_dense)
+
+    w_dense, v_dense = splin.eigsh(A_dense, M=M_dense, k=3, v0=v0)
+    w, v = splin.eigsh(A, M=M_dense, k=3, v0=v0)
+
+    assert_allclose(w, w_dense)
+    assert_allclose(v, v_dense)
+
+
+def test_svds():
+    np.random.seed(1234)
+
+    A_dense = np.random.rand(10, 10)
+    A_dense = A_dense @ A_dense.T
+    A = sparse.COO(A_dense)
+
+    v0 = np.random.rand(10)
+
+    u0, s0, vt0 = splin.svds(A_dense, k=3, v0=v0)
+    u, s, vt = splin.svds(A, k=3, v0=v0)
+
+    assert_allclose(s, s0)
+    assert_allclose(u, u0)
+    assert_allclose(vt, vt0)
+
+
+def test_lobpcg():
+    np.random.seed(1234)
+
+    A_dense = np.random.rand(10, 10)
+    A_dense = A_dense @ A_dense.T
+    A = sparse.COO(A_dense)
+
+    X = np.random.rand(10, 3)
```

Comment:
Please also add here an alternative test with
`X = np.random.randn(10, 1)`
- LOBPCG defaults to the standard 'eig' solver if the size[1] of X is 5 times larger than size[0]